Composer/packages/client/src/pages/publish/Publish.tsx

```diff
@@ -200,286 +228,169 @@ const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: st
-          types={publishTypes}
-          updateSettings={updatePublishTarget}
-        />
-      ),
+    setBotPublishHistoryList(newPublishHistory);
+  };
+  const updateSelectedBots = (selectedBots) => {
+    const bots: IBotStatus[] = [];
+    selectedBots.forEach((bot) => {
+      bots.push({
+        id: bot.id,
+        name: bot.name,
+        publishTarget: bot.publishTarget,
+        publishTargets: bot.publishTargets,
+      });
     });
-  }, [editTarget, publishTypes, updatePublishTarget]);
-
-  const rollbackToVersion = useMemo(
-    () => async (version) => {
-      const sensitiveSettings = getSensitiveProperties(settings);
-      await rollbackToVersionDispatcher(projectId, selectedTarget, version.id, sensitiveSettings);
-    },
-    [projectId, selectedTarget]
-  );
-
-  const publish = useMemo(
-    () => async (comment) => {
-      // publish to remote
-      if (selectedTarget && settings.publishTargets) {
-        if (settings.qna && Object(settings.qna).subscriptionKey) {
-          await setQnASettings(projectId, Object(settings.qna).subscriptionKey);
-        }
-        const sensitiveSettings = getSensitiveProperties(settings);
-        await publishToTarget(projectId, selectedTarget, { comment: comment }, sensitiveSettings);
-
-        // update the target with a lastPublished date
-        const updatedPublishTargets = settings.publishTargets.map((profile) => {
-          if (profile.name === selectedTarget.name) {
-            return {
-              ...profile,
-              lastPublished: new Date(),
-            };
-          } else {
-            return profile;
-          }
-        });
-
-        await setPublishTargets(updatedPublishTargets, projectId);
-      }
-    },
-    [projectId, selectedTarget, settings.publishTargets]
-  );
-
-  const onEdit = async (index: number, item: PublishTarget) => {
-    const newItem = { item: item, index: index };
-    setEditTarget(newItem);
-    setEditDialogHidden(false);
+    setSelectedBots(bots);
   };
-
-  const onDelete = useMemo(
-    () => async (index: number) => {
-      const result = await OpenConfirmModal(
-        formatMessage('This will delete the profile. Do you wish to continue?'),
-        null,
-        {
-          confirmBtnText: formatMessage('Yes'),
-          cancelBtnText: formatMessage('Cancel'),
+  const publish = async (items: IBotStatus[]) => {
+    // notifications
+    setShowNotifications(
+      items.reduce((accumulator, item) => {
+        accumulator[item.id] = true;
+        return accumulator;
+      }, {})
+    );
+    const notification = createNotification(getPendingNotificationCardProps(items));
+    setPendingNotification(notification);
+    addNotification(notification);
+
+    // publish to remote
+    if (items.length > 0) {
+      for (const bot of items) {
+        if (bot.publishTarget && bot.publishTargets) {
```

Comment:
should the name be bot.currentTarget && bot.publishTargets?

Is bot.publishTarget just the currently selected target?

Reply:
Sure